ui/src/app/pages/data/data-set/export-dataset-modal.tsx

```diff
@@ -110,27 +110,29 @@ export const ExportDatasetModal = ({
     );
   };
 
-  const createExportDatasetRequest = (): DataSetExportRequest => {
+  const createExportDatasetRequest = (
+    language: AnalysisLanguage
+  ): DataSetExportRequest => {
     return {
       dataSetRequest: createDataSetRequest(),
-      kernelType,
+      analysisLanguage: language,
```

Comment:
use a parameter instead of state, in order to better control state transitions.  This helps to fix a race condition bug.